apps/mobile/app/_layout.tsx

```diff
@@ -15,22 +15,25 @@ export const unstable_settings = {
 }
 
 export default function HomeLayout() {
-  const pathname = usePathname()
-  const router = useRouter()
+  const navigation = useNavigationContainerRef()
 
   useEffect(() => {
-    const route = storage.getString('navRoute')
-    if (route && route !== '/') {
-      setTimeout(() => {
-        router.push(route)
-      }, 500)
+    if (__DEV__) {
+      const state = storage.getString('navRoute-state')
+      if (state) {
+        const [, ...routes] = JSON.parse(state)
+        setTimeout(() => {
+          for (const route of routes) {
+            navigation.dispatch(CommonActions.navigate(route))
+          }
+        }, 500)
```

Comment:
man this hack is so crazy. I wish we had real state persistence with Expo Router, and its so sad that we have an error when this value is much less than 500ms

Reply:
replaced with requestAnimationFrame tho we will have to revisit later with React Navigation.